.github/CONTRIBUTING.md

```diff
@@ -75,6 +75,114 @@ Also, have you signed the [Contribution License Agreement](https://cla.microsoft
 
 **If your contribution is going to be a major effort, you should give us a heads-up first. We have a lot of items captured in our backlog and we release every two weeks, so before you spend the time, just check with us to make sure your plans and ours are in sync :) Just open an issue on GitHub and tag it as "contribution".**
 
+#### Commit Guidelines
+We have very precise rules over how our git commit messages can be formatted. This leads to more readable messages that are easy to follow when looking through the project history.
+
+**Commit Message Format**
+
+Each commit message consists of a header, a body (optional) and a footer (optional). The header has a special format that includes a type, a scope and a subject:
+
+```C#
+
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+
+```
+The header is mandatory and the body and footer are optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier to read on GitHub as well as in various git tools.
+
+Footer should contain a [closing reference](#https://help.github.com/articles/closing-issues-using-keywords/) to an issue if any.
+
+**Revert**
+
+If the commit reverts a previous commit, it should begin with revert:, followed by the header of the reverted commit. In the body it should say: This reverts commit <hash>., where the hash is the SHA of the commit being reverted.
+
+**Rebase and Squash**
+
+* Its manadatory to squash all your commits per scope (i.e package). It is also important to rebase your commits on master.
+* Optionally you can split your commits on the basis of the package you are providing code to.
+
+**Type**
+
+Must be one of the following:
+
+* build: Changes that affect the build system
+* docs: Documentation only changes (eg ReadMe)
+* feat: A new feature
+* fix: A bug fix
+* perf: A code change that improves performance
+* refactor: A code change that neither fixes a bug nor adds a feature
+
+Optionally you could also use the following scope:
+
+* style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+* test: Adding missing tests or correcting existing tests
+
+**Scope**
+
+The scope should be the name of the package affected as perceived by person reading changelog generated from commit messages.
+
+    * shared
+    * devices-client
```

Comment:
To maintain parity with dps, this should be called `iothub-client`, `iothub-service`.

Reply:
Sounds good!